src/Controller/BaseController.php

```diff
@@ -0,0 +1,93 @@
+<?php declare(strict_types=1);
+
+namespace Becklyn\RadBundle\Controller;
+
+use Becklyn\RadBundle\Exception\EntityRemovalBlockedException;
+use Becklyn\RadBundle\Exception\LabeledEntityRemovalBlockedException;
+use Becklyn\RadBundle\Form\FormErrorMapper;
+use Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException;
+use Psr\Log\LoggerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Base class for all controllers
+ */
+abstract class BaseController extends AbstractController
```

Comment:
If you are thinking of downloading a file, there already is the ->file() method in Symfony‘s Controller trait. 

If you are thinking of providing a string as file download, then I would leave it in a trait, because a) that logic is pretty encapsulated and doesn’t use any services and b) it isn’t used very often. 

We need to look at the Controller class as a namespace, where we don’t want to clutter the available method names with hundreds of helpers.

Reply:
The problem with `$this->file()` is that you'd have to generate the file name yourself. In our most recent implementation we're automatically removing non-ascii characters to prevent Symfony from crashing. With a more friendly version of `$this->file()` we could automate this step.

Reply:
->file() is also automatically generating the fallback name, take a look at the implementation.

Reply:
I did. It's either using the name of the physical file or the one you're passing in. The fallback is the one you're passing in, which needs to be normalized.

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php#L136-L149

Reply:
Yes, but you still need to click into the code.

The called method `BinaryFileResponse::setContentDisposition()` will then set a fallback name, if no explicit fallback is provided and the regular file name contains invalid characters, right here:

https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/HttpFoundation/BinaryFileResponse.php#L162-L174